packages/toolkit/src/components/card-pipeline-hub/Head.tsx

```diff
@@ -0,0 +1,136 @@
+"use client";
+
+import * as React from "react";
+import {
+  InstillStore,
+  OrganizationOwner,
+  Pipeline,
+  UserOwner,
+  toastInstillError,
+  useDeleteUserPipeline,
+  useInstillStore,
+  useShallow,
+} from "../../lib";
+import { Icons, useToast } from "@instill-ai/design-system";
+import { EntityAvatar } from "../EntityAvatar";
+import { useRouter } from "next/navigation";
+
+export const HeadSkeleton = () => {
+  return (
+    <div className="flex w-full flex-row gap-x-2 p-3">
+      <div className="my-auto h-8 w-8 shrink-0 grow-0 animate-pulse rounded-full bg-semantic-bg-secondary" />
+      <div className="h-[38px] w-20 animate-pulse rounded bg-semantic-bg-secondary"></div>
+    </div>
+  );
+};
+
+const selector = (store: InstillStore) => ({
+  accessToken: store.accessToken,
+  enabledQuery: store.enabledQuery,
+});
+
+export const Head = ({
+  ownerID,
+  isOwner,
+  pipeline,
+}: {
+  ownerID: string;
+  isOwner: boolean;
+  pipeline: Pipeline;
+}) => {
+  const router = useRouter();
+  const { toast } = useToast();
+  const { accessToken } = useInstillStore(useShallow(selector));
+
+  const deletePipeline = useDeleteUserPipeline();
+  async function handleDeletePipeline() {
+    try {
+      await deletePipeline.mutateAsync({
+        pipelineName: pipeline.name,
+        accessToken: accessToken ? accessToken : null,
+      });
```

Comment:
When handle the mutation, we should block the whole process when the accessToken is null

The reason we mark the accessToken as nullable string is to have flexibility on working with some edge case 🦅